evm/gateway.js

```diff
@@ -278,6 +278,34 @@ async function processCommand(config, chain, options) {
             }
         }
 
+        case 'isContractCallApproved': {
+            const payloadHash = keccak256(arrayify(payload));
```

Comment:
Just take payload hash as input directly. Easier to get from the amplifier message